### PR TITLE
waypipe: init at 0.6.1

### PR DESCRIPTION
**Pull request by primeos**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
  - Works with `weston-terminal` and `firefox`, but e.g. `alacritty` and `qutebrowser` don't work yet (qutebrowser might be fixable with environment variables, but I haven't tried it yet).
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/applications/networking/remote/waypipe/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    # Optional dependencies:
+    mesa lz4 zstd ffmpeg_4 libva
+  ];
+
+  enableParallelBuilding = true;
+
+  mesonFlags = [ "-Dwerror=false" ]; # TODO: Report warnings upstream
+
+  meta = with stdenv.lib; {
+    description = "Network transparency with Wayland";
```
Comment (gilligan) [inline pkgs/applications/networking/remote/waypipe/default.nix]: This really is a minor detail but this description just states some goal but doesn't actually capture what this package is about? How about something more like

> network proxy for wayland client applications

WDYT?

**Review by gilligan — APPROVED** [review]

File: pkgs/applications/networking/remote/waypipe/default.nix (hunk 1 above)
Comment (primeos) [reply pkgs/applications/networking/remote/waypipe/default.nix]: Good idea, thanks :) While we're at it I've als added a `longDescription` from the README. 

Additionally I've added `-unstable` to `pname`. It is usable but results vary depending on the application and upstream notes that a lot of things might change / it's in early development.

File: pkgs/applications/networking/remote/waypipe/default.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  mesonFlags = [ "-Dwerror=false" ]; # TODO: Report warnings upstream
+
+  meta = with stdenv.lib; {
+    description = "A network proxy for Wayland clients (applications)";
+    longDescription = ''
+      waypipe is a proxy for Wayland clients. It forwards Wayland messages and
+      serializes changes to shared memory buffers over a single socket. This
+      makes application forwarding similar to ssh -X feasible.
+    '';
+    homepage    = https://mstoeckl.com/notes/gsoc/blog.html;
```
Comment (disassembler) [inline pkgs/applications/networking/remote/waypipe/default.nix]: don't put the extra spaces here.

**Review by disassembler — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/remote/waypipe/default.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "waypipe-unstable";
+  version = "0.6.1";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.freedesktop.org";
+    owner = "mstoeckl";
+    repo = pname;
```
Comment (dywedir) [inline pkgs/applications/networking/remote/waypipe/default.nix]:
```suggestion
    repo = "waypipe";
```

Comment (primeos) [reply pkgs/applications/networking/remote/waypipe/default.nix]: Will this be an improvement? IMO this is subjective (or do we have guidelines regarding this?) and AFAIK we currently mix both variants in Nixpkgs. I'm fine with either version but tbh it feels a bit like bikeshedding.

File: pkgs/applications/networking/remote/waypipe/default.nix (hunk 2 above)
Comment (primeos) [reply pkgs/applications/networking/remote/waypipe/default.nix]: Removed it (was for alignment but that doesn't work well with `longDescription` and it's probably a bad idea in general).

File: pkgs/applications/networking/remote/waypipe/default.nix (hunk 3 above)
Comment (dywedir) [reply pkgs/applications/networking/remote/waypipe/default.nix]: I mean, pname is set to "waypipe-unstable". So we should use `repo = "waypipe"` for now.

Comment (primeos) [reply pkgs/applications/networking/remote/waypipe/default.nix]: Oh, sorry! My bad, I've completely missed that and was surprised about the change :o :D

Thanks for paying better attention!

(Note: I'll keep `unstable` in `pname` for now as this is afaik the preferred solution, see https://github.com/NixOS/nixpkgs/pull/67460#discussion_r317415142)

**Comment by dywedir** [comment]
Thanks!

**Pull request merged**